airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/adapters.py

```diff
@@ -219,6 +269,9 @@ def __init__(self, stream: Stream, _slice: Optional[Mapping[str, Any]], message_
         self._stream = stream
         self._slice = _slice
         self._message_repository = message_repository
+        self._sync_mode = sync_mode
+        self._cursor_field = cursor_field
```

Comment:
> Q1: can we use the Stream's cursor_field?

The typing between the two is different:
* catalog_cursor_field: Optional[List[str]]
* stream: cursor_field: Union[str, List[str]]

[The previous code was using the catalog one](https://github.com/airbytehq/airbyte/blob/ad1f3f7a7a4dad61edafdb3be6dd471e18193632/airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py#L247). Hence I wanted to avoid breaking changes.

> Could they ever be different?

I see two cases:
* Given the typing, yes. I assume that the logic to convert one to the other is consigned to `Stream._wrapped_cursor_field`. We could maybe use `stream.as_airbyte_stream().default_cursor_field`
* There is nothing preventing anyone from providing a different catalog than what the code returns as a cursor. In practice, I assume this does not happen but in theory, this would be a breaking change

> Q2: should this assert that the cursor field is not nested (by verifying it only has one element). This is not a requirement for for Stream but we make that assertion in ThreadBasedConcurrentStream

Do you mean `assumption` or `assertion`? I don't see the assertion in `ThreadBasedConcurrentStream`. Is this a fair assumption? I feel like I've already been told by why can't we support nested cursor fields? In any case, I don't mind being explicit about it in `ThreadBasedConcurrentStream.__init__`, something like:

```
if cursor_field and not isinstance(cursor_field, str):
  raise ValueError("Nested cursor fields are not supported by ThreadBasedConcurrentStream because <reason>")
```